test/core.jl

```diff
@@ -7140,3 +7140,10 @@ end
 struct SplatBadIterate; end
 Base.iterate(s::SplatBadIterate, args...) = ()
 @test_throws BoundsError (SplatBadIterate()...,)
+
+# Issue #34206/34207
+function mre34206(a)
+    b = ntuple(_ -> view(a, :), 1)[1]
```

Comment:
```suggestion
function mre34206(a, n)
    va = view(a, :)
    b = ntuple(_ -> va, n)::Tuple{Vararg{typeof(va)}}
    return b[1].offset1
end
@test mre34206([44], 1) == 0
```

The test was relying on constant propagation heuristics being disabled because we decide not to inline that ntuple call. I think this rewrite should give us greater confidence that changes to inference would find it very difficult to cause this to get any better or worse type information than we need it to have to trigger this codegen path.